squad/core/models.py

```diff
@@ -853,6 +853,25 @@ def get_regressions(self):
     def get_fixes(self):
         return self.__get_yaml_field__(self.fixes)
 
+    def get_exceeded_thresholds(self):
+        # Return a list of all (threshold, metric) objects for those
+        # thresholds that were exceeded by corresponding metrics.
+        thresholds_exceeded = []
+        for threshold in self.build.project.metricthreshold_set.all():
```

Comment:
I have a feeling this for loop can be replaced with annotated queryset. How is the performance here with big data sets?

Reply:
I optimized it a bit. I was banging my head a log with how to do some sort of join here with prefetch_related but no dice. Please review again and tell me what you think.